### ISSUE TITLE
job error with JWD tmp directory on 22.01

### ISSUE THREAD

**Issue by bgruening** [galaxyproject/galaxy#13902]
The tool `devteam/concat/gops_concat_1/1.0.1` fails reproducible on EU with
```
rm: cannot remove '/data/jwd/main/046/709/46709070/tmp/out.471762': No such file or directory
rm: cannot remove '/data/jwd/main/046/709/46709070/tmp/err.471762': No such file or directory
```

It works on ORG, I assume it has something to do with `metadata_strategy: 'extended'`

```
sn06:/data/jwd/main/046/709/46709070$ tree

├── _configs
├── configs
├── galaxy_46709070.condor.log
├── galaxy_46709070.e
├── galaxy_46709070.ec
├── galaxy_46709070.o
├── galaxy_46709070.sh
├── home
├── __instrument_cgroup__metrics
├── __instrument_core_epoch_end
├── __instrument_core_epoch_start
├── __instrument_core_galaxy_memory_mb
├── __instrument_core_galaxy_slots
├── __instrument_hostname_hostname
├── memory_statement.log
├── metadata
│   ├── metadata_kwds_output
│   ├── metadata_out_output
│   ├── metadata_override_output
│   ├── metadata_results_output
│   ├── object_store_conf.json
│   ├── outputs_new
│   │   ├── collections_attrs.txt
│   │   ├── datasets_attrs.txt
│   │   ├── datasets_attrs.txt.provenance
│   │   ├── export_attrs.txt
│   │   ├── jobs_attrs.txt
│   │   ├── libraries_attrs.txt
│   │   └── tool.xml
│   ├── outputs_populated
│   │   ├── collections_attrs.txt
│   │   ├── datasets_attrs.txt
│   │   ├── datasets_attrs.txt.provenance
│   │   ├── export_attrs.txt
│   │   ├── implicit_collection_jobs_attrs.txt
│   │   ├── jobs_attrs.txt
│   │   └── libraries_attrs.txt
│   ├── params.json
│   ├── registry.xml
│   └── set.py
├── _outputs
│   └── galaxy_dataset_4181b347-eee1-43c3-a87b-5ee7abcb0679.dat
├── outputs
│   ├── galaxy_dataset_4181b347-eee1-43c3-a87b-5ee7abcb0679.dat
│   ├── tool_stderr
│   └── tool_stdout
├── tmp
├── tmp.20220507-191546
│   ├── err.471762
│   └── out.471762
├── tool_script.sh
├── _working
└── working
    └── conda_activate.log

12 directories, 42 files
```

**Comment by mvdbeek** [comment]
That's the FIFO files for redirected stdout and stderr. Has the job been resubmitted ? You can actually see those in `tmp.20220507-191546`

**Comment by bgruening** [comment]
I doubled checked with @gmauro  and we do not think it gets resubmitted.

> You can actually see those in tmp.20220507-191546

yes, see the above `tree` output.

**Comment by gmauro** [comment]
this is what we have in the log:
```
May 11 10:39:07 sn06.galaxyproject.eu python[3066673]: galaxy.jobs.handler DEBUG 2022-05-11 10:39:07,026 [pN:handler_key_1,p:3066673,tN:JobHandlerQueue.monitor_thread] Grabbed Job(s): 46911833
May 11 10:39:07 sn06.galaxyproject.eu python[3066673]: galaxy.jobs.mapper DEBUG 2022-05-11 10:39:07,282 [pN:handler_key_1,p:3066673,tN:JobHandlerQueue.monitor_thread] (46911833) Mapped job to destination id: 1cores_4.0G
May 11 10:39:07 sn06.galaxyproject.eu python[3066673]: galaxy.jobs.handler DEBUG 2022-05-11 10:39:07,317 [pN:handler_key_1,p:3066673,tN:JobHandlerQueue.monitor_thread] (46911833) Dispatching to condor runner
May 11 10:39:07 sn06.galaxyproject.eu python[3066673]: galaxy.jobs DEBUG 2022-05-11 10:39:07,334 [pN:handler_key_1,p:3066673,tN:JobHandlerQueue.monitor_thread] (46911833) Persisting job destination (destination id: 1cores_4.0G)
May 11 10:39:07 sn06.galaxyproject.eu python[3066673]: galaxy.objectstore DEBUG 2022-05-11 10:39:07,366 [pN:handler_key_1,p:3066673,tN:JobHandlerQueue.monitor_thread] Using preferred backend 'files12' for creation of Job 46911833
May 11 10:39:09 sn06.galaxyproject.eu python[3066673]: galaxy.jobs DEBUG 2022-05-11 10:39:09,910 [pN:handler_key_1,p:3066673,tN:JobHandlerQueue.monitor_thread] (46911833) Working directory for job is: /data/jwd/main/046/911/46911833
May 11 10:39:10 sn06.galaxyproject.eu python[3066673]: galaxy.jobs.runners DEBUG 2022-05-11 10:39:10,017 [pN:handler_key_1,p:3066673,tN:JobHandlerQueue.monitor_thread] Job [46911833] queued (2700.246 ms)
May 11 10:39:10 sn06.galaxyproject.eu python[3066673]: galaxy.jobs.handler INFO 2022-05-11 10:39:10,028 [pN:handler_key_1,p:3066673,tN:JobHandlerQueue.monitor_thread] (46911833) Job dispatched
May 11 10:39:10 sn06.galaxyproject.eu python[3066673]: galaxy.jobs DEBUG 2022-05-11 10:39:10,319 [pN:handler_key_1,p:3066673,tN:CondorRunner.work_thread-7] Job wrapper for Job [46911833] prepared (276.109 ms)
May 11 10:39:10 sn06.galaxyproject.eu python[3066673]: galaxy.jobs.command_factory INFO 2022-05-11 10:39:10,576 [pN:handler_key_1,p:3066673,tN:CondorRunner.work_thread-7] Built script [/data/jwd/main/046/911/46911833/tool_script.sh] for tool command [[ "$(basename "$CONDA_DEFAULT_ENV")" = "$(basename '/usr/local/tools/_conda/envs/mulled-v1-a4cb0f5781d6f9de5e288400a2ac9fa34202068e5723b723249629522600dc67')" ] || {
May 11 10:39:10 sn06.galaxyproject.eu python[3066673]: } ; python '/opt/galaxy/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/concat/32e1c8dac438/concat/gops_concat.py' '/data/dnb06/galaxy_db/files/0/2/e/dataset_02ec4df6-0132-4787-8939-e9b09a2df04a.dat' '/data/dnb06/galaxy_db/files/3/d/8/dataset_3d8a33e6-7c66-4a2b-8074-f115b3d2c2e1.dat' '/data/jwd/main/046/911/46911833/outputs/galaxy_dataset_6c63c6b8-1616-44f2-905c-c34c19f7806f.dat' -1 1,2,3,0 -2 1,2,3,0 --sameformat]
May 11 10:39:11 sn06.galaxyproject.eu python[3066673]: galaxy.jobs.runners DEBUG 2022-05-11 10:39:11,487 [pN:handler_key_1,p:3066673,tN:CondorRunner.work_thread-7] (46911833) command is: mkdir -p working outputs configs
May 11 10:39:11 sn06.galaxyproject.eu python[3066673]: tee -a '../outputs/tool_stderr' < "$__err" >&2 & /bin/bash /data/jwd/main/046/911/46911833/tool_script.sh > "$__out" 2> "$__err"; return_code=$?; echo $return_code > /data/jwd/main/046/911/46911833/galaxy_46911833.ec; cd '/data/jwd/main/046/911/46911833';
May 11 10:39:11 sn06.galaxyproject.eu python[3066673]: galaxy.jobs.runners.condor DEBUG 2022-05-11 10:39:11,693 [pN:handler_key_1,p:3066673,tN:CondorRunner.work_thread-7] (46911833) submitting file /data/jwd/main/046/911/46911833/galaxy_46911833.sh
May 11 10:39:11 sn06.galaxyproject.eu python[3066673]: galaxy.jobs.runners.condor INFO 2022-05-11 10:39:11,942 [pN:handler_key_1,p:3066673,tN:CondorRunner.work_thread-7] (46911833) queued as 31935884
May 11 10:39:26 sn06.galaxyproject.eu python[3066673]: galaxy.jobs.runners.condor DEBUG 2022-05-11 10:39:26,899 [pN:handler_key_1,p:3066673,tN:CondorRunner.monitor_thread] (46911833/31935884) job is now running
May 11 10:39:36 sn06.galaxyproject.eu python[3066673]: galaxy.jobs.runners.condor DEBUG 2022-05-11 10:39:36,260 [pN:handler_key_1,p:3066673,tN:CondorRunner.monitor_thread] (46911833/31935884) job is now running
May 11 10:39:46 sn06.galaxyproject.eu python[3066673]: galaxy.jobs.runners.condor DEBUG 2022-05-11 10:39:46,626 [pN:handler_key_1,p:3066673,tN:CondorRunner.monitor_thread] (46911833/31935884) job is now running
May 11 10:39:46 sn06.galaxyproject.eu python[3066673]: galaxy.jobs.runners.condor DEBUG 2022-05-11 10:39:46,662 [pN:handler_key_1,p:3066673,tN:CondorRunner.monitor_thread] (46911833/31935884) job has completed
May 11 10:39:48 sn06.galaxyproject.eu python[3066673]: galaxy.jobs INFO 2022-05-11 10:39:48,189 [pN:handler_key_1,p:3066673,tN:CondorRunner.work_thread-2] Collecting metrics for Job 46911833 in /data/jwd/main/046/911/46911833
May 11 10:39:50 sn06.galaxyproject.eu python[3066673]: galaxy.jobs DEBUG 2022-05-11 10:39:50,346 [pN:handler_key_1,p:3066673,tN:CondorRunner.work_thread-2] job_wrapper.finish for job 46911833 executed (2664.403 ms)

```

**Comment by mvdbeek** [comment]
@gmauro @bgruening can you try https://github.com/mvdbeek/galaxy/commit/853ef69d0240010a7a9816bccd9185313a6b771e ?

**Comment by gmauro** [comment]
done and still failing but not showing `rm: cannot remove ...` message

**Comment by gmauro** [comment]
to add more details, the state of the job in the db is `ok` but the job's step in the history is red

**Comment by gmauro** [comment]
and the job working directory has been removed from Galaxy, so that means the job succeeded.

**Comment by mvdbeek** [comment]
I don't think the error message is related to that. What's recorded for tool_stdout, tool_stderr, job_stdout, job_stderr and exit code ?

**Comment by gmauro** [comment]
tool_stdout, tool_stderr, job_stdout are empty
job_stderr has the `rm: cannot remove ...` message
exit code is 0

**Comment by mvdbeek** [comment]
That's with https://github.com/mvdbeek/galaxy/commit/853ef69d0240010a7a9816bccd9185313a6b771e ?

**Comment by bgruening** [comment]
You commit fixed the `rm: connect remove` stuff. But the dataset is still red. It seems the job is in `ok` state and the dataset is in `error` state.

**Comment by mvdbeek** [comment]
There should be a structured message in the log telling you the failure reason. Can you find that one ?

**Comment by bgruening** [comment]
I don't see any error:

```
>journalctl -u galaxy-handler@* --since '10 minutes ago' --no-pager | grep 46951242

May 12 14:44:42 sn06.galaxyproject.eu python[2457775]: galaxy.jobs.handler DEBUG 2022-05-12 14:44:42,261 [pN:handler_key_3,p:2457775,tN:JobHandlerQueue.monitor_thread] Grabbed Job(s): 46951242
May 12 14:44:47 sn06.galaxyproject.eu python[2457775]: galaxy.jobs.mapper DEBUG 2022-05-12 14:44:47,646 [pN:handler_key_3,p:2457775,tN:JobHandlerQueue.monitor_thread] (46951242) Mapped job to destination id: 1cores_4.0G
May 12 14:44:47 sn06.galaxyproject.eu python[2457775]: galaxy.jobs.handler DEBUG 2022-05-12 14:44:47,708 [pN:handler_key_3,p:2457775,tN:JobHandlerQueue.monitor_thread] (46951242) Dispatching to condor runner
May 12 14:44:47 sn06.galaxyproject.eu python[2457775]: galaxy.jobs DEBUG 2022-05-12 14:44:47,725 [pN:handler_key_3,p:2457775,tN:JobHandlerQueue.monitor_thread] (46951242) Persisting job destination (destination id: 1cores_4.0G)
May 12 14:44:49 sn06.galaxyproject.eu python[2457775]: galaxy.objectstore DEBUG 2022-05-12 14:44:49,563 [pN:handler_key_3,p:2457775,tN:JobHandlerQueue.monitor_thread] Using preferred backend 'files12' for creation of Job 46951242
May 12 14:44:51 sn06.galaxyproject.eu python[2457775]: galaxy.jobs DEBUG 2022-05-12 14:44:51,403 [pN:handler_key_3,p:2457775,tN:JobHandlerQueue.monitor_thread] (46951242) Working directory for job is: /data/jwd/main/046/951/46951242
May 12 14:44:51 sn06.galaxyproject.eu python[2457775]: galaxy.jobs.runners DEBUG 2022-05-12 14:44:51,414 [pN:handler_key_3,p:2457775,tN:JobHandlerQueue.monitor_thread] Job [46951242] queued (3705.207 ms)
May 12 14:44:51 sn06.galaxyproject.eu python[2457775]: galaxy.jobs.handler INFO 2022-05-12 14:44:51,420 [pN:handler_key_3,p:2457775,tN:JobHandlerQueue.monitor_thread] (46951242) Job dispatched
May 12 14:44:51 sn06.galaxyproject.eu python[2457775]: galaxy.jobs DEBUG 2022-05-12 14:44:51,934 [pN:handler_key_3,p:2457775,tN:CondorRunner.work_thread-3] Job wrapper for Job [46951242] prepared (502.474 ms)
May 12 14:44:55 sn06.galaxyproject.eu python[2457775]: galaxy.jobs.command_factory INFO 2022-05-12 14:44:55,131 [pN:handler_key_3,p:2457775,tN:CondorRunner.work_thread-3] Built script [/data/jwd/main/046/951/46951242/tool_script.sh] for tool command [[ "$(basename "$CONDA_DEFAULT_ENV")" = "$(basename '/usr/local/tools/_conda/envs/mulled-v1-a4cb0f5781d6f9de5e288400a2ac9fa34202068e5723b723249629522600dc67')" ] || {
May 12 14:44:55 sn06.galaxyproject.eu python[2457775]: } ; python '/opt/galaxy/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/concat/32e1c8dac438/concat/gops_concat.py' '/data/dnb06/galaxy_db/files/7/c/5/dataset_7c5b20f9-b523-4621-b270-cb7d7dd9e18a.dat' '/data/dnb06/galaxy_db/files/7/c/5/dataset_7c5b20f9-b523-4621-b270-cb7d7dd9e18a.dat' '/data/jwd/main/046/951/46951242/outputs/galaxy_dataset_6bd6c42e-38cb-4523-860d-fba596a406bb.dat' -1 1,2,3,0 -2 1,2,3,0 --sameformat  && sleep 10]
May 12 14:45:00 sn06.galaxyproject.eu python[2457775]: galaxy.jobs.runners DEBUG 2022-05-12 14:45:00,910 [pN:handler_key_3,p:2457775,tN:CondorRunner.work_thread-3] (46951242) command is: mkdir -p working outputs configs
May 12 14:45:00 sn06.galaxyproject.eu python[2457775]: tee -a '../outputs/tool_stderr' < "$__err" >&2 & /bin/bash /data/jwd/main/046/951/46951242/tool_script.sh > "$__out" 2> "$__err"; return_code=$?; echo $return_code > /data/jwd/main/046/951/46951242/galaxy_46951242.ec; cd '/data/jwd/main/046/951/46951242';
May 12 14:45:02 sn06.galaxyproject.eu python[2457775]: galaxy.jobs.runners.condor DEBUG 2022-05-12 14:45:02,112 [pN:handler_key_3,p:2457775,tN:CondorRunner.work_thread-3] (46951242) submitting file /data/jwd/main/046/951/46951242/galaxy_46951242.sh
May 12 14:45:05 sn06.galaxyproject.eu python[2457775]: galaxy.jobs.runners.condor INFO 2022-05-12 14:45:05,510 [pN:handler_key_3,p:2457775,tN:CondorRunner.work_thread-3] (46951242) queued as 31972630
May 12 14:45:15 sn06.galaxyproject.eu python[2457775]: galaxy.jobs.runners.condor DEBUG 2022-05-12 14:45:15,987 [pN:handler_key_3,p:2457775,tN:CondorRunner.monitor_thread] (46951242/31972630) job is now running
May 12 14:45:22 sn06.galaxyproject.eu python[2457775]: galaxy.jobs.runners.condor DEBUG 2022-05-12 14:45:22,896 [pN:handler_key_3,p:2457775,tN:CondorRunner.monitor_thread] (46951242/31972630) job is now running
May 12 14:48:54 sn06.galaxyproject.eu python[2457775]: galaxy.jobs.runners.condor DEBUG 2022-05-12 14:48:54,708 [pN:handler_key_3,p:2457775,tN:CondorRunner.monitor_thread] (46951242/31972630) job is now running
May 12 14:48:54 sn06.galaxyproject.eu python[2457775]: galaxy.jobs.runners.condor DEBUG 2022-05-12 14:48:54,844 [pN:handler_key_3,p:2457775,tN:CondorRunner.monitor_thread] (46951242/31972630) job has completed
May 12 14:49:01 sn06.galaxyproject.eu python[2457775]: galaxy.jobs INFO 2022-05-12 14:49:01,269 [pN:handler_key_3,p:2457775,tN:CondorRunner.work_thread-6] Collecting metrics for Job 46951242 in /data/jwd/main/046/951/46951242
May 12 14:49:01 sn06.galaxyproject.eu python[2457775]: galaxy.jobs DEBUG 2022-05-12 14:49:01,288 [pN:handler_key_3,p:2457775,tN:CondorRunner.work_thread-6] job_wrapper.finish for job 46951242 executed (1455.214 ms)
``

**Comment by mvdbeek** [comment]
Ah, right, because of extended_metadata. There should be some more information in the `outputs_populated` directory.

**Comment by bgruening** [comment]
```
galaxy@sn06:/data/jwd/main/046/951/46951242/metadata/outputs_populated$ jq --color-output . jobs_attrs.txt 
[
  {
    "command_line": "#!/bin/bash\n\n# The following block can be used by the job system\n# to ensure this script is runnable before actually attempting\n# to run it.\nif [ -n \"$ABC_TEST_JOB_SCRIPT_INTEGRITY_XYZ\" ]; then\n    exit 42\nfi\n[ \"$(basename \"$CONDA_DEFAULT_ENV\")\" = \"$(basename '/usr/local/tools/_conda/envs/mulled-v1-a4cb0f5781d6f9de5e288400a2ac9fa34202068e5723b723249629522600dc67')\" ] || {\nMAX_TRIES=3\nCOUNT=0\nwhile [ $COUNT -lt $MAX_TRIES ]; do\n    . '/usr/local/tools/_conda/bin/activate' '/usr/local/tools/_conda/envs/mulled-v1-a4cb0f5781d6f9de5e288400a2ac9fa34202068e5723b723249629522600dc67' > conda_activate.log 2>&1\n    if [ $? -eq 0 ];then\n        break\n    else\n        let COUNT=COUNT+1\n        if [ $COUNT -eq $MAX_TRIES ];then\n            echo \"Failed to activate conda environment! Error was:\"\n            cat conda_activate.log\n            exit 1\n        fi\n        sleep 10s\n    fi\ndone\n} ; [ \"$(basename \"$CONDA_DEFAULT_ENV\")\" = \"$(basename '/usr/local/tools/_conda/envs/mulled-v1-a4cb0f5781d6f9de5e288400a2ac9fa34202068e5723b723249629522600dc67')\" ] || {\nMAX_TRIES=3\nCOUNT=0\nwhile [ $COUNT -lt $MAX_TRIES ]; do\n    . '/usr/local/tools/_conda/bin/activate' '/usr/local/tools/_conda/envs/mulled-v1-a4cb0f5781d6f9de5e288400a2ac9fa34202068e5723b723249629522600dc67' > conda_activate.log 2>&1\n    if [ $? -eq 0 ];then\n        break\n    else\n        let COUNT=COUNT+1\n        if [ $COUNT -eq $MAX_TRIES ];then\n            echo \"Failed to activate conda environment! Error was:\"\n            cat conda_activate.log\n            exit 1\n        fi\n        sleep 10s\n    fi\ndone\n} ; python '/opt/galaxy/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/concat/32e1c8dac438/concat/gops_concat.py' '/data/dnb06/galaxy_db/files/7/c/5/dataset_7c5b20f9-b523-4621-b270-cb7d7dd9e18a.dat' '/data/dnb06/galaxy_db/files/7/c/5/dataset_7c5b20f9-b523-4621-b270-cb7d7dd9e18a.dat' '/data/jwd/main/046/951/46951242/outputs/galaxy_dataset_6bd6c42e-38cb-4523-860d-fba596a406bb.dat' -1 1,2,3,0 -2 1,2,3,0 --sameformat  && sleep 10",
    "create_time": "2022-05-12T12:44:42.038037",
    "exit_code": null,
    "galaxy_version": null,
    "id": 46951242,
    "implicit_output_dataset_collection_mapping": {},
    "info": null,
    "input_dataset_collection_element_mapping": {},
    "input_dataset_collection_mapping": {},
    "input_dataset_mapping": {
      "input1": [
        106954799
      ],
      "input2": [
        106954799
      ]
    },
    "job_messages": [],
    "job_stderr": null,
    "job_stdout": null,
    "model_class": "Job",
    "output_dataset_collection_mapping": {},
    "output_dataset_mapping": {
      "output": [
        107450641
      ]
    },
    "params": {
      "__input_ext": "bed",
      "chromInfo": "/opt/galaxy/tool-data/shared/ucsc/chrom/?.len",
      "dbkey": "?",
      "input1": {
        "values": [
          {
            "id": 106954799,
            "src": "hda"
          }
        ]
      },
      "input2": {
        "values": [
          {
            "id": 106954799,
            "src": "hda"
          }
        ]
      },
      "sameformat": "true"
    },
    "state": "ok",
    "tool_id": "toolshed.g2.bx.psu.edu/repos/devteam/concat/gops_concat_1/1.0.1",
    "tool_stderr": null,
    "tool_stdout": null,
    "tool_version": "1.0.1",
    "traceback": null,
    "update_time": "2022-05-12T12:44:51.919503"
  }
]
galaxy@sn06:/data/jwd/main/046/951/46951242/metadata/outputs_populated$ tree
.
├── collections_attrs.txt
├── datasets_attrs.txt
├── datasets_attrs.txt.provenance
├── export_attrs.txt
├── implicit_collection_jobs_attrs.txt
├── jobs_attrs.txt
└── libraries_attrs.txt

0 directories, 7 files
```

Not much useful in there ...

**Comment by bgruening** [comment]
```
[
  {
    "annotation": "",
    "blurb": "4 regions",
    "copied_from_history_dataset_association_id_chain": [],
    "create_time": "None",
    "dataset": {
      "_extra_files_path": null,
      "created_from_basename": null,
      "deleted": false,
      "external_filename": null,
      "file_size": 0,
      "hashes": [],
      "id": 88436992,
      "model_class": "Dataset",
      "object_store_id": "files12",
      "purged": false,
      "sources": [],
      "state": "ok",
      "total_size": 0,
      "uuid": "6bd6c42e-38cb-4523-860d-fba596a406bb"
    },
    "dataset_uuid": "6bd6c42e-38cb-4523-860d-fba596a406bb",
    "deleted": false,
    "designation": "output",
    "extension": "bed",
    "hid": null,
    "id": 107450641,
    "info": "",
    "metadata": {
      "chromCol": 1,
      "column_names": [],
      "column_types": [
        "str",
        "int",
        "int"
      ],
      "columns": 3,
      "comment_lines": 0,
      "data_lines": 4,
      "dbkey": "?",
      "delimiter": "\t",
      "endCol": 3,
      "nameCol": null,
      "startCol": 2,
      "strandCol": 0,
      "viz_filter_cols": [
        4
      ]
    },
    "model_class": "HistoryDatasetAssociation",
    "name": "Concatenate on data 476",
    "peek": "chr\t1\t10\nchr\t4\t20\nchr\t1\t10\nchr\t4\t20\n",
    "state": "ok",
    "tags": [],
    "tool_version": null,
    "update_time": "None",
    "validated_state": "unknown",
    "validated_state_message": null,
    "visible": true
  }
]
```
What? State=ok?

**Comment by mvdbeek** [comment]
To update what we found out on gitter:

There is a code file with the following content:
```python
def exec_after_process(app, inp_data, out_data, param_dict, tool=None, stdout=None, stderr=None):
    """Verify the output data after each run"""
    for data in out_data.values():
        try:
            if stderr and len( stderr ) > 0:
                raise Exception( stderr )
        except Exception:
            data.blurb = JOB_ERROR
            data.state = JOB_ERROR
```
this is of course a terrible way to do it and should be removed, and I think that might be the culprit here. Not sure though why `stderr` contains content, it seems to be null when it reaches the metadata script.

**Comment by natefoo** [comment]
Regarding the fix in #13905, should we just convert that `rm` to `rm -f` instead of redirecting stderr?

**Comment by nsoranzo** [comment]
> Regarding the fix in #13905, should we just convert that `rm` to `rm -f` instead of redirecting stderr?

Agreed. Same applies to https://github.com/galaxyproject/galaxy/pull/13913 .

**Comment by mvdbeek** [comment]
I'm gonna close this, the remaining part of this will be fixed with https://github.com/galaxyproject/galaxy/pull/13642 and the tool itself should drop the use of the codefile.
